File: 0422-valid-word-square/0422-valid-word-square.java
class Solution {
    public boolean validWordSquare(List<String> words) {
        // for(int i = 0; i < words.size(); ++i)
        //     for(int j = 0; j < words.get(i).length(); ++j) {
        //         if(i >= words.get(i).length() || j >= words.size()) return false;
        //         char c1 = words.get(i).charAt(j);
        //         char c2 = words.get(j).charAt(i);
        //         if(c1 != c2) return false;
        //     }
        // return true;
                int n = words.size();
        for(int i=0; i<n; i++){
            for(int j=0; j<words.get(i).length(); j++){
                if(j >= n || words.get(j).length() <= i || words.get(j).charAt(i) != words.get(i).charAt(j))
                    return false;
            }
        }
        return true;
    }
}

File: 0422-valid-word-square/0422-valid-word-square.cpp
class Solution {
public:
    bool validWordSquare(vector<string>& words) {
        for(int i=0;i<words.size();i++)// words.size()  is used  to find vector size 
        {
            if(words[i].size()>words.size())// suppose words[i] string length is greater than the vector size then that string not preent in that . so  we return the false.
            {
                return false;
            }
            for(int j=0;j<words[i].size();j++)
            {
                if(words[i][j]!=words[j][i])// eg: ["abcd","bnrt","crm","dt"]      we compare words[0][3]==words[3][0]
                {
                    return false;
                }
            }
        }
        return true;
    }
};